src/main/scala/mesosphere/marathon/core/launchqueue/impl/LaunchQueueActor.scala

```diff
@@ -116,12 +116,19 @@ private[impl] class LaunchQueueActor(
 
   private[this] def receiveTaskUpdateToSuspendedActor: Receive = {
     case taskChanged: TaskChanged if suspendedLauncherPathIds(taskChanged.appId) =>
-      deferMessageToSuspendedActor(taskChanged, taskChanged.appId)
+      // Do not defer. If an AppTaskLauncherActor restarts, it retrieves a new task list.
+      // If we defer this, there is a potential deadlock (resolved by timeout):
+      //   * AppTaskLauncher waits for in-flight tasks
+      //   * TaskOp gets processed and one of the update steps calls this here
+      //   * ... blocked until timeout ...
+      //   * The task launch notification (that the AppTaskLauncherActor waits for) gets sent to the actor
+      sender() ! None
```

Comment:
this only gets invoked after an app purge. If there is the necessity, a new AppTaskLaunchActor is started afterwards again. That new actor will load all app tasks again, so it doesn't matter that it missed some update in the past.